test/standalone/dependencyFromBuildZig/build.zig

```diff
@@ -0,0 +1,16 @@
+const std = @import("std");
+
+pub fn build(b: *std.Build) void {
+    const test_step = b.step("test", "Test it");
+    b.default_step = test_step;
+
+    const dep1 = b.dependency("other", .{});
+
+    const build_runner = @import("root");
+    const deps = build_runner.dependencies;
+    const zon_decls = @typeInfo(deps.packages).Struct.decls;
+    const pkg = @field(deps.packages, zon_decls[0].name);
+    const dep2 = b.dependencyFromBuildZig(pkg.build_zig, .{});
```

Comment:
It doesn't really matter for this test because in the end it still tests the same end result, but this reflection should be able to be reduced to a simple `b.dependencyFromBuildZig(@import("other", .{}))`.